File: frontend/vite/src/models/patient.ts
import { IDistrict, IProvince } from "./location";
import { IClient, IClientContact } from "./client";

export interface IPatient {
    uid: number;
    clientPatientId: string;
    patientId: string;
    firstName: string;
    middleName: string;
    lastName: string;
    client: IClient;
    clientUid: number;
    gender?: string | number;
    age: number;
    dateOfBirth: Date;
    ageDobEstimated: boolean;
    phoneHome: string;
    phoneMobile: string;
    consentSms: boolean;
    district: IDistrict;
    districtUid: number;
    province: IProvince;
    provinceUid: number;
    country: IDistrict;
    countryUid: number;
}

File: backend/felicity_lims/felicity/tests/utils/patient.py
add_patient_query = """
  mutation AddPatient($payload: PatientInputType!){
  createPatient(payload: $payload) {
    ... on PatientType {
        uid
        clientPatientId
        firstName
        lastName
        age
        gender
        dateOfBirth
        clientUid
        phoneMobile
        consentSms   
    }
    ... on OperationError {
        error
    }
  }
}
"""
